src/lib/broadcast.jl

```diff
@@ -263,8 +263,8 @@ end
 
   end
 
-  @adjoint CUDA.CuArray{N,T}(xs::Array) where {N,T} =
-    CUDA.CuArray{N,T}(xs), Δ -> (convert(Array, Δ), )
+  @adjoint CUDA.CuArray{N,T,B}(xs::Array) where {N,T,B} =
```

Comment:
Maybe try something like `(::Type{<:CuArray{T,N}})(...)` to prevent locking into specific typevars again?